array_api_tests/test_elementwise_functions.py

```diff
@@ -24,106 +25,106 @@
 from .test_broadcasting import broadcast_shapes
 
 
+@pytest.mark.parametrize("expr", ["xp.abs(x)", "abs(x)"])
 @given(xps.arrays(dtype=xps.numeric_dtypes(), shape=hh.shapes()))
-def test_abs(x):
-    if dh.is_int_dtype(x.dtype):
-        minval = dh.dtype_ranges[x.dtype][0]
-        if minval < 0:
-            # abs of the smallest representable negative integer is not defined
-            mask = xp.not_equal(x, ah.full(x.shape, minval, dtype=x.dtype))
-            x = x[mask]
-    a = xp.abs(x)
-    ph.assert_shape("abs", a.shape, x.shape)
-    assert ah.all(ah.logical_not(ah.negative_mathematical_sign(a))), "abs(x) did not have positive sign"
+def test_abs(expr, x):
+    if x.dtype in dh.int_dtypes:
+        # abs of the smallest representable negative integer is not defined
+        m, _ = dh.dtype_ranges[x.dtype]
+        mask = xp.not_equal(x, ah.full(x.shape, m, dtype=x.dtype))
+        x = x[mask]
+    out = eval(expr)
```

Comment:
It would be better to explicitly pass a globals namespace to eval, like `eval(expr, {'xp': xp})`. That way any names that are imported or defined in this file can't accidentally affect it.